lib/active_merchant/billing/gateways/mobilexpress.rb

```diff
@@ -97,11 +99,15 @@ def add_authentication(xml, options={})
       end
 
       def add_customer_data(xml, options)
-        xml.CustomerID options[:customer_id]
+        xml.CustomerID options[:customer_id] unless empty?(options[:customer_id])
         xml.CustomerName options[:customer_name] unless empty?(options[:customer_name])
         xml.ClientIP options[:ip] unless empty?(options[:ip])
       end
 
+      def add_transaction(xml, options)
+        xml.TransactionId (options[:order_id] || SecureRandom.hex(10))
```

Comment:
I take it `:order_id` isn’t one of the options you need to do a `empty?` check on.

Reply:
Exactly ... in this case, I am just opting for:

> If there's no `:order_id` in the options then send a unique value

Because the `TransactionId` needs to be unique per transaction so better always send it with a random string.